### PR TITLE
Add option to show labels in Google Workspace user creation component

### PR DESCRIPTION
**Pull request by daledupreez**
#### Changes proposed in this Pull Request

* This PR adds a new `showLabels` prop to the `GSuiteNewUserList` and `GSuiteNewUser` components that control whether each input field should include a label. The new prop defaults to `false` to retain backwards compatibility.
* The PR also enables these labels for the email comparison page, and toggles the equivalent flag for WordPress.com Email.
* The PR also fixes the vertical alignment for the trash icon in the add new mailbox page for WordPress.com Email.

##### Positive test

* Navigate to Upgrades -> Domains, and find a domain without email added, where the domain also has DNS managed at WordPress.com _and_ hasn't had Google purchased for the domain yet.
* Click on the "Add" button in the Email column for that domain
* Verify that the WordPress.com Email card shows labels for all the form fields
* Click on the "Add Google Workspace" button to expand the Google Workspace card
* Verify that the Google Workspace card shows labels for all the form fields

_Note: If you can't see Google Workspace as an option, you'll need to try a different domain._ 

##### Negative test

* Navigate to Upgrades -> Domains
* Click on the "Add a domain" button/drop-down in the top right corner.
* Click on the "to this site" option
* Select a domain
* Verify that you see the Google Workspace upsell page, and the form inputs display without any labels

##### Test for Add new mailbox page

* Navigate to Upgrades -> Domains
* Select a domain where you already have a WordPress.com Email subscription (buy a new one if needed)
  - If you purchased a new domain, click on the "Manage email" button after checkout.
  - If you had the domain already, select the domain and then click on "Manage your email accounts".
* Click on "Add New Mailbox"
* Verify that the trash icon appears vertically aligned with the input field to its left

##### Before

_WordPress.com Email without labels_
Screenshot 2021-04-12 at 10 50 37

_Google Workspace without labels_
Screenshot 2021-04-12 at 10 52 57

##### After

_WordPress.com Email with labels_
Screenshot 2021-04-13 at 21 59 35

_Google Workspace with labels_
Screenshot 2021-04-13 at 22 00 11

### CODE REVIEW

**Comment by daledupreez** [comment]
@saygunnyc, could you take a look at the visuals for this PR to verify that it is moving towards your intended design?

I think it would also be good to call out any specific improvements we can take on (other than hiding the "Remove" button for the first item in the list!)

**Comment by saygunnyc** [comment]
Thank you, Dale! The only comment I have regarding this PR in particular would be the vertical spacings in between labels and the input fields. Can we add in 4px spacing in between?

I believe I've shared feedback for things not related to this PR. But happy to further discuss if you need clarification.

**Comment by daledupreez** [comment]
Thanks for taking a look, @saygunnyc!

The spacing issue is a structural problem -- I was using
```
<label>
  Some label
  <input ... />
</label>
```
instead of
```
<label for="...">Some label</label>
<input ... />
```

**Comment by daledupreez** [comment]
@saygunnyc, as a quick note here, I checked and the default margin between the label and the input is 5px, so I've replicated that spacing -- [reference](https://github.com/Automattic/wp-calypso/blob/22c01ccea0625b55c0e01b3bd13a492bc2a44215/client/components/forms/form-label/style.scss#L5).
I also noticed that the text colour had been overridden, so I've reinstated the standard text colour for the form controls.

**Comment by saygunnyc** [comment]
Wonderful. Thank you, Dale! I really appreciate you checking the original spacings. We should only change the current library **intentionally.**

File: client/components/gsuite/gsuite-new-user-list/new-user.tsx [hunk 1]
```diff
@@ -1,28 +1,49 @@
 /**
  * External dependencies
  */
+import classnames from 'classnames';
```
Comment (olaseni) [inline client/components/gsuite/gsuite-new-user-list/new-user.tsx]: This should probably be imported as `classNames` for consistency with camelCase rules

File: client/components/gsuite/gsuite-new-user-list/new-user.tsx [hunk 2]
```diff
@@ -103,47 +128,56 @@ const GSuiteNewUser: FunctionComponent< Props > = ( {
...
 
 	return (
 		<div className="gsuite-new-user-list__new-user">
 			<FormFieldset>
-				<div className="gsuite-new-user-list__new-user-section">
+				<div
+					className={ classnames(
+						'gsuite-new-user-list__new-user-section',
+						showLabels ? 'show-labels' : null
+					) }
```
Comment (olaseni) [inline client/components/gsuite/gsuite-new-user-list/new-user.tsx]: It's probably simpler to use `classNames` in this format.

```suggestion
					className={ classNames(
						'gsuite-new-user-list__new-user-section',
						{  'show-labels' :  showLabels }
					) }
```

File: client/components/gsuite/gsuite-new-user-list/style.scss [hunk 3]
```diff
@@ -57,15 +42,51 @@
...
+		}
+	}
+
+	&.show-labels {
+		.gsuite-new-user-list__new-user-remove-user-button {
+			@include breakpoint-deprecated( '>800px' ) {
+				margin-top: calc( 1.5em + 5px );
+			}
+		}
+	}
```
Comment (olaseni) [inline client/components/gsuite/gsuite-new-user-list/style.scss]: Could we maybe make this more readable and possibly save on the braces if we reduced the nesting?

```suggestion
	&.show-labels  .gsuite-new-user-list__new-user-remove-user-button {
			@include breakpoint-deprecated( '>800px' ) {
				margin-top: calc( 1.5em + 5px );
			}
	}
```

**Review by olaseni — APPROVED** [review]
This works well as described and depicted!

I'll approve this. However I have some non-blocking revision suggestions which should be penned in after the fact.

File: client/components/gsuite/gsuite-new-user-list/new-user.tsx (hunk 1 above)
Comment (daledupreez) [reply client/components/gsuite/gsuite-new-user-list/new-user.tsx]: Done in both locations where this was imported as `classnames`.

File: client/components/gsuite/gsuite-new-user-list/new-user.tsx (hunk 2 above)
Comment (daledupreez) [reply client/components/gsuite/gsuite-new-user-list/new-user.tsx]: Thanks for the suggestion!
I adopted that format and realised this class should be added one level up.

File: client/components/gsuite/gsuite-new-user-list/style.scss (hunk 3 above)
Comment (daledupreez) [reply client/components/gsuite/gsuite-new-user-list/style.scss]: I took care of this partly by changing the element that `show-labels` is attached to, partly by moving this into the same section as the label CSS, and by shifting the CSS to one less nesting level.

**Pull request merged**